#controller_manager加载的控制器配置（通过参数服务器配置）
controller:
  sentry_chassis_controller:
    type: sentry_chassis_controller/SentryChassisController
    # 8个关节的名称（需和URDF中的关节名一致）
    joint_names: ["left_front_wheel_joint", "right_front_wheel_joint", "left_back_wheel_joint", "right_back_wheel_joint", 
                "left_front_pivot_joint", "right_front_pivot_joint", "left_back_pivot_joint", "right_back_pivot_joint"]
    # 8个关节的PID参数
    pid:
      left_front_wheel_joint:  {p: 7.0, i: 0.1, d: 0.05, i_max: 0.5, i_min: -0.5}
      right_front_wheel_joint: {p: 7.0, i: 0.1, d: 0.05, i_max: 0.5, i_min: -0.5}
      left_back_wheel_joint:   {p: 7.0, i: 0.1, d: 0.05, i_max: 0.5, i_min: -0.5}
      right_back_wheel_joint:  {p: 7.0, i: 0.1, d: 0.05, i_max: 0.5, i_min: -0.5}
      left_front_pivot_joint:  {p: 2.0, i: 0.2, d: 0.1, i_max: 1.0, i_min: -1.0}
      right_front_pivot_joint: {p: 2.0, i: 0.2, d: 0.1, i_max: 1.0, i_min: -1.0}
      left_back_pivot_joint:   {p: 2.0, i: 0.2, d: 0.1, i_max: 1.0, i_min: -1.0}
      right_back_pivot_joint:  {p: 2.0, i: 0.2, d: 0.1, i_max: 1.0, i_min: -1.0}

  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50




